tests/unit_tests/training/utils/test_train_utils.py

```
@@ -1540,6 +1541,64 @@ def test_report_throughput_negative_elapsed_wct(self, mock_print_rank_0):
         assert f"iteration {iteration}" in warning_message
 
+    @mock.patch("megatron.bridge.training.utils.train_utils.print_rank_0")
+    def test_report_throughput_resume_from_ckpt(self, mock_print_rank_0):
+        global_batch_size = 128
+        micro_batch_size = 2
+        iteration = 100
+        seq_length = 8192
+        window_size = 10
+
+        # first run
+        history_wct = [i + random.uniform(2, 2.5) for i in range(window_size)]
```

Ruff (0.15.1)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)